lib/screens/route_patterns/route_pattern.ex

```diff
@@ -55,21 +55,66 @@ defmodule Screens.RoutePatterns.RoutePattern do
 
     case get_json_fn.("route_patterns", params) do
       {:ok, result} ->
-        stop_sequences =
-          get_in(result, [
-            "included",
-            Access.filter(&(&1["type"] == "trip")),
-            "relationships",
-            "stops",
-            "data",
-            Access.all(),
-            "id"
-          ])
+        {:ok, get_stop_sequences_from_result(result)}
 
-        {:ok, stop_sequences}
+      _ ->
+        :error
+    end
+  end
+
+  @doc """
+  Gets the list of stop sequences for stop and creates a map of platform IDs to parent station name.
+  Assumes that all stop sequences in result are platforms.
+  """
+  @spec fetch_parent_station_sequences_through_stop(Stop.id(), list(String.t())) ::
+          {:ok, list(list(Stop.id())), map()} | :error
+  def fetch_parent_station_sequences_through_stop(
+        stop_id,
+        route_filters,
+        get_json_fn \\ &V3Api.get_json/2
+      ) do
+    params = %{
+      "include" => "representative_trip.stops,route",
+      "filter[stop]" => stop_id,
+      "filter[route]" => Enum.join(route_filters, ",")
```

Comment:
Previously, route_filters was a conditional parameter, but here it's required. Which is fine, I think? Because all the cases where we want to use it is for subway. In that case, does it need to be a parameter at all? We could set it right here in this function to :light_rail and :subway.

Reply:
I thought about this too, and I decided to leave it because the filters passed in here are more specific than `:light_rail` and `:subway`. It's actually the routes that service the station. So to keep the V3 query as filtered as possible, I think it's best we keep it as a parameter.